tests/benchmark/README.md

```diff
@@ -0,0 +1,320 @@
+
+# The benchmarks directory 
+This directory contains the following:
+* **Header** files with the classes that are used to define the benchmark test routines. See [Fixtures](#fixtures) section for details.   
+It also includes `bm_definitions.h` header that contains benchmarks utils such as:
+    - `IndexType` struct
+    - common typedefs
+    - macro shortcuts to the static data members of `BM_VecSimGeneral` and `BM_VecSim_Index<index_type_t>`
+* **subdirectory bm_initialization** - Each file is associated with a test set (basics/batch etc) of a particular **data type**, and fits for both single and multi indices. The purpose  of these files is to avoid code replications. Here we define and register the tests using Google benchmarks macros.  See [How to define and register a new test](#how-to-define-and-register-a-new-test) section.
+* **subdirectory run_files** - contains a file for each combination of: set of benchmark tests, index data type, index type(single/multi).  
+In this file we:
+    - Initialize **BM_VecSimGeneral static data members** such as is_multi, dim, n_vectors, hnsw_index_file, etc, according to the set of parameters of the index loaded from the specified file.
+    - In addition, the macro `BM_FUNC_NAME(bm_func, algo)` is defined to be used in the initialization file. The tests' names are used to generate the graphs on the Grafana dashboard. If two tests share the same name, they will appear on the same graph. Hence, This macro has to **define unique benchmark tests names**. (see [Tests naming](#tests-naming))
+    - Include the bm_initialization file that calls google benchmark definitions and registrations macros.
+    - Call `BENCHMARK_MAIN()` 
+
+
+## How to define and register a new test
+Google benchmarks library supports templated fixture (see [Google benchmarks user guide](https://github.com/google/benchmark/blob/main/docs/user_guide.md#templated-benchmarks) regarding this topic).  
+
+### Definition:  
+`BENCHMARK_TEMPLATE_DEFINE_F(fixture_name, test_name, data type)(benchmark::State &st){fixture_name::test_implementation(st, test_args);}`  
+In the definition we:
+1. Bind the test to a specific fixture
+2. Give it a unique name (see [Tests naming](#tests-naming))
+3. Specify the fixture data type (currently available: `fp32_index_t` and `fp64_index_t`) (see [Fixture template type](#fixture-template-type).  
+4. Call the function, defined in the fixture class, that implements the required test case.
+### Registration
+`BENCHMARK_REGISTER_F(test_fixture, test_name)->Args({val1, val2})->ArgNames({"name1", "name2"})->Iterations(val0)->Unit(benchmark::kMillisecond)`  
+Here we register to the benchmarks running loop the test that was bind to `test_fixture` and has `test_name` .  
+*NOTE:* The **registration order** is important. Changing it might violate the ability to compare between one benchmark run to another. The reason to that is that we have tests (such as addLabel and DeleteLabel) that modify the hnsw graph.
```

Comment:
```suggestion
*NOTE:* The **registration order** is important. Changing it might violate the ability to compare one benchmark run to another. We have tests (such as addLabel and DeleteLabel) that modify the indices and are not idempotent.
```